ocaml/CMakeLists.txt

```diff
@@ -43,16 +45,19 @@ add_custom_command(
   DEPENDS ${OCAMLC}
   COMMENT "Compiling ocamlbuild")
 
+add_custom_command(
+  OUTPUT ${OPAM_SRC}
```

Comment:
please make sure that this isn't going to re-download it every time you run 'make'

If there isn't a clear way to do it, the usual way is something along the lines of:
- make `fetch_and_setup_src.sh` also touch `opam.stamp` in the build directory
- add a build rule to always produce `opam.stamp` which depends on `fetch_and_setup_src.sh`

Reply:
actually, that problem already exists, and we already rebuild. This should ideally be fixed, but it's not really making things any worse at the moment.

Reply:
Do you think something along the lines of
```
diff --git a/ocaml/CMakeLists.txt b/ocaml/CMakeLists.txt
index d105b2d..13919bc 100644
--- a/ocaml/CMakeLists.txt
+++ b/ocaml/CMakeLists.txt
@@ -74,6 +74,12 @@ add_custom_command(
   DEPENDS ${OPAM_SRC}
   COMMENT "Compiling opam")
 
+add_custom_command(
+    OUTPUT ${OCAML_PREFIX}/opam.stamp
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+    DEPENDS ${OPAM}
+    COMMENT "Prevent multiple rebuild of opam")
+
 set(OCAMLC_FOUND TRUE)
 
 add_custom_target(ocaml
diff --git a/ocaml/fetch_and_setup_src.sh b/ocaml/fetch_and_setup_src.sh
index d210cd5..eadcb06 100755
--- a/ocaml/fetch_and_setup_src.sh
+++ b/ocaml/fetch_and_setup_src.sh
@@ -20,3 +20,5 @@ shasum -a 256 -c checksum
 rm -rf opam
 tar xzf "${OPAM_PKG}"
 mv "opam-full-${OPAM_VERSION}" opam
+mkdir -p build
+touch build/opam.stamp
```
would work ? I added a custom command for opam.stamp without actual commands, and updated the bash script to touch the file.

Reply:
My bad. opam seems to be rebuilding more than once when I tried the out of source scenario. I'll prepare a "stamp" commit to fix that

Reply:
As a matter of fact, I'll have to start working on that once the third-party repo is cleaned (updated to opam2 and ocamlbuild/ocaml removed). So that won't be part of this PR.